sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java

```diff
@@ -135,4 +135,28 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
                 return resultSet.getObject(columnIndex);
         }
     }
+
+    private static Object getIntTypeValue(final ResultSet resultSet, final int columnIndex) throws SQLException {
+        ResultSetMetaData metaData = resultSet.getMetaData();
+        if (metaData.isSigned(columnIndex)) {
+            return resultSet.getInt(columnIndex);
+        }
+        long value = resultSet.getLong(columnIndex);
+        if (value > Integer.MAX_VALUE) {
+            return value;
+        }
+        return (int) value;
+    }
+
+    private static Object getBigIntTypeValue(final ResultSet resultSet, final int columnIndex) throws SQLException {
+        ResultSetMetaData metaData = resultSet.getMetaData();
+        if (metaData.isSigned(columnIndex)) {
+            return resultSet.getLong(columnIndex);
+        }
+        BigDecimal value = resultSet.getBigDecimal(columnIndex);
+        if (new BigDecimal(Long.MAX_VALUE).compareTo(value) <= -1) {
+            return value;
+        }
```

Comment:
Use BigInteger replace BigDecimal may be better.

can use getBigDecimal().toBigInteger to get BigInteger